cmake/functions/four_c_configure_dependency.cmake

```
@@ -5,7 +5,180 @@
+        _supported_external_version
+        )
+      # Sanity check: is the supported version a major.minor version?
+      if(NOT _supported_internal_version MATCHES "^[0-9]+\\.[0-9]+$")
+        message(
+          FATAL_ERROR
+            "File 'dependencies/supported_version/${_package_name}.txt' contains an unsupported version format: ${_supported_internal_version}. "
+            "Needs to be of the form 'major.minor'."
+          )
+      endif()
+      if(${_index} EQUAL 0)
+        set(_latest_version ${_supported_internal_version})
+      endif()
+
+      # Check if the hash matches the beginning of the first element as we might have a short _sha.
+      string(FIND ${_supported_sha} ${_sha} _sha_position)
+      if(${_sha_position} EQUAL 0)
+        message(
+          STATUS
+            "Identified 4C-internal version ${_supported_internal_version} of ${_package_name} (hash ${_sha}, package's own version name: ${_supported_external_version})."
+          )
+        if(${_index} EQUAL 0)
+          message(STATUS "This is the latest supported version of ${_package_name}.")
+        else()
+          message(
+            STATUS
+              "NOTE: This is not the latest supported version of ${_package_name}. Check the file 'dependencies/supported_version/${_package_name}.txt' for more information."
+            )
+        endif()
+        set(FOUR_C_${_package_name_sanitized}_INTERNAL_VERSION
+            ${_supported_internal_version}
+            PARENT_SCOPE
+            )
+
+        string(REGEX MATCH "^([0-9]+)\\.([0-9]+)$" _dummy ${_supported_internal_version})
+        set(_major ${CMAKE_MATCH_1})
+        set(_minor ${CMAKE_MATCH_2})
+
+        set(FOUR_C_${_package_name_sanitized}_INTERNAL_VERSION
+            ${_supported_internal_version}
+            PARENT_SCOPE
+            )
+        set(FOUR_C_${_package_name_sanitized}_INTERNAL_VERSION_MAJOR
+            ${_major}
+            PARENT_SCOPE
+            )
+        set(FOUR_C_${_package_name_sanitized}_INTERNAL_VERSION_MINOR
+            ${_minor}
+            PARENT_SCOPE
+            )
+        return()
+      endif()
+      math(EXPR _index "${_index} + 1")
+    endforeach()
+    message(STATUS "")
+    message(STATUS "Could not identify your version of ${_package_name} with hash ${_sha}.")
+    message(STATUS "Supported versions are:")
+    foreach(_supported_version ${_supported_versions})
+      separate_arguments(_supported_version_split UNIX_COMMAND ${_supported_version})
+      list(
+        POP_FRONT
+        _supported_version_split
+        _supported_sha
+        _supported_internal_version
+        _supported_external_version
+        )
+      message(
+        STATUS
+          "  - ${_supported_internal_version} (hash ${_supported_sha}, package's own version name: ${_supported_external_version})"
+        )
+    endforeach()
+    message(
+      STATUS "Assuming your version is newer than the latest supported version ${_latest_version}."
```

Above ("Could not identify ...") the message is an English sentence in first person without naming the first person.
```suggestion
      STATUS "Assume your version is newer than the latest supported version ${_latest_version}."
```